app/models/detector/citation.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+class Detector
+  # Detector::Citation attempts to identify citations based on the prevalence of individual sub-patterns. It is not
+  # targeted at a particular citation format, but was designed based on characteristics of five formats: APA, MLA,
+  # Chicago, Terabian, and IEEE.
+  #
+  # It receives a Term object, which is parsed in various ways en route to calculating a final score. Terms with a
+  # higher score are more citation-like, while a score of 0 indicates a Term that has no hallmarks of being a citation.
+  # Terms whose score is higher than the REQUIRED_SCORE value can be registered as a Detection.
+  class Citation
+    attr_reader :score, :subpatterns, :summary
+
+    # Citation patterns are regular expressions which attempt to identify structures that are part of many citations.
+    # This object is used as part of the pattern_checker method. Some of these patterns may get promoted to the Detector
+    # model if they prove useful beyond a Citation context.
+    CITATION_PATTERNS = {
+      apa_volume_issue: /\d+\(\d+\)/,
+      no: /no\.\s\d+/,
+      pages: /\d+-+\d+/,
+      pp: /pp\.\s\d+/,
+      vol: /vol\.\s\d+/,
+      year_parens: /\(\d{4}\)/,
+      brackets: /\[.*?\]/,
+      lastnames: /[A-Z][a-z]+[.,]/,
+      quotes: /&quot;.*?&quot;/
+    }.freeze
+
+    # The required score value is the threshold needed for a Term to be officially recorded with a Detection.
+    REQUIRED_SCORE = 6
+
+    # Summary thresholds are used by the calculate_score method. This class counts the number of occurrences of specific
+    # characters in the @summary instance variable. The thresholds here determine whether any of those counts are high
+    # enough to contribute to the Term's citation score.
+    SUMMARY_THRESHOLDS = {
+      characters: 25,
+      colons: 2,
+      commas: 2,
+      periods: 2,
+      semicolons: 2,
+      words: 5
+    }.freeze
+
+    # The initializer handles the parsing of a Term object, and subsequent population of the @subpatterns, @summary,
+    # and @score instance variables. @subpatterns contains all the citation components which have been flagged by the
+    # CITATION_PATTERNS hash. @summary contains counts of how often certain characters or words appear in the Term.
+    # Finally, the @score value is a summary of how many elements in the subpatterns or summary report were detected.
+    #
+    # @note This method can be called directly via Detector::Citation.new(Term). It is also called indirectly via the
+    #       Detector::Citation.record(Term) instance method. This method can be called directly when a Detection is not
+    #       desired.
+    def initialize(term)
```

Comment:
I've been thinking about a future state of having tacos run in "not logging mode". Having our detectors work based on `Strings` instead of `Terms` would be key to that potential functionality.

The `.record` methods could still use the `Term` as those are explicitly part of our write mode processes.

I'm sharing this just as information on some future direction we may head. If you prefer to leave this initialize method using the Term that is fine. Some of our Detectors use Terms and some use Strings. Eventually I want to normalize that as it is a type of technical debt to have them work differently for now good reason. As we haven't formally picked a path forward, either Term is String is acceptable, but I'm leaning towards proposing we adopt Strings for all Detectors even if we use Terms for `record` to give us the flexibility checking how TACOS would respond to input without having to store that input. The main use cases I have for that desired feature is using the playground without having to store my test queries as Terms as well as potentially opening up TACOS for others to experiment with without us introducing a privacy concern storing data from other systems not in our control. The latter is not planned so the former is what I'd like to decide how to handle.

Reply:
I support having a path through the application to allow for querying with strings, but I'd prefer to make that change in a coordinated way. Of the detectors we have now, every example that uses an `initialize` method (StandardIdentifiers, LCSH, and now Citation) does so based on a Term method - so I'd prefer to keep this consistency. I worry about having a state, even temporarily, where one detector behaves differently from the others.

As part of such a change, I wonder about how to name the methods to clarify what is expected, i.e. something like:

```
Detector::Citation.new(phrase) # using built-in "new" as a method

Detector::Citation.register_term(term) # method named to clarify expected argument, instead of "record"
```

But, we can have this discussion more thoroughly in whatever ticket centers that refactoring.

Reply:
Yup, that makes sense to me.